src/common/openingTime.ts

```diff
@@ -0,0 +1,150 @@
+import { strDash, strOpeningTimeRest } from "./constants";
+import { dayOfWeekToText } from "./datetime";
+
+type OpeningTimeRange = {
+  beg: string;
+  end: string;
+};
+
+type OpeningTimeRangesDaysOfWeek = Array<Array<OpeningTimeRange>>;
+
+export function getOpeningStatus(daysTimes: Array<Array<string>>): {
+  isOpen: boolean;
+  isOpen24Hour: boolean;
+  closedTime: string;
+  nextOpenTime: string;
+  hasOpeningTime: boolean;
+} {
+  const now = new Date();
+  const daysTimesFormatted = formatOpeningDaysTimes(daysTimes);
+
+  return {
+    isOpen: isOpenAtTime(daysTimesFormatted, now),
+    isOpen24Hour: isOpen24HoursAtTime(daysTimesFormatted, now),
+    closedTime: getClosedTime(daysTimesFormatted, now),
+    nextOpenTime: getNextOpenTime(daysTimesFormatted, now),
+    hasOpeningTime: hasOpeningTimeOnDayOfWeek(daysTimesFormatted, now.getDay()),
+  };
+}
+
+function formatOpeningDaysTimes(
+  daysTimes: Array<Array<string>>
+): OpeningTimeRangesDaysOfWeek {
+  return daysTimes.map((dayTimes: Array<string>) => {
+    return dayTimes
+      .filter((timeRange: string) => {
+        return timeRange !== strOpeningTimeRest;
+      })
+      .map((timeRange: string) => {
+        const [beg, end] = timeRange.split(strDash);
+        return {
+          beg: beg,
+          end: end,
+        };
+      });
+  });
+}
+
+function getClosedTime(
+  openingTimeRangesDaysOfWeek: OpeningTimeRangesDaysOfWeek,
+  time: Date
+): string {
+  const dayOfWeek = time.getDay();
+  const timeStr = getTimeStr(time);
+
+  for (const timeRange of openingTimeRangesDaysOfWeek[dayOfWeek]) {
+    if (isInTimeRange(timeStr, timeRange)) {
+      return timeRange.end;
+    }
+  }
+
+  return "";
+}
+
+function getNextOpenTime(
+  openingTimeRangesDaysOfWeek: OpeningTimeRangesDaysOfWeek,
+  time: Date
+): string {
+  const dayOfWeek = time.getDay();
+  const timeStr = getTimeStr(time);
+
+  for (const timeRange of openingTimeRangesDaysOfWeek[dayOfWeek]) {
+    if (timeStr < timeRange.beg) {
+      return timeRange.beg;
+    }
+  }
+
+  for (let i = 1; i < 7; ++i) {
+    const anotherDayOfWeek = (dayOfWeek + i) % 7;
+
+    const timeRanges = openingTimeRangesDaysOfWeek[anotherDayOfWeek];
+    if (timeRanges.length > 0) {
+      return dayOfWeekToText(anotherDayOfWeek, false) + timeRanges[0].beg;
+    }
+  }
+
+  for (const timeRange of openingTimeRangesDaysOfWeek[dayOfWeek]) {
```

Comment:
Not sure why we have to do another loop, same as the beginning of "getNextOpenTime".

Reply:
This loop checks the passed opening time today for some store which opening only one day in a week. Although there is no case in our current data, I prepare for it. By the way, `麵屋清流` has the minimum opening days, 2 in our current data.

Reply:
Got it! Would it be better if the previous loop starts at i=1 and ends at 7?
Also, consider that the open time is 17:00-20:00 every Wednesday, and the current time is 19:00 on Wednesday.
It seems more clear to show the next open time `Wednesday 17:00` than `17:00`.

Reply:
I will change the ending of previous loop to 7. And I made a mistake that the second loop of `openingTimeRangesDaysOfWeek[dayOfWeek]` should return the first beginning time instead of checking. Thank you!

Reply:
Great.